### ISSUE TITLE
I believe my  calculator should pass all (but one just discovered bug) tests but only 7 of 16 pass.

### ISSUE THREAD

**Issue by MikePia** [freeCodeCamp/testable-projects-fcc#1179]
#### Issue Description

Running on codepen [here]( https://codepen.io/mpete/pen/OJMxEjy?editors=1111). I am using the inset tests from [bundle.js](https://cdn.freecodecamp.org/testable-projects-fcc/v1/bundle.js) placed in a script tag on the html page. Can reproduce it by clicking on the testSuite menu, choosing javascript Calculator and clicking run tests.  The failures, I think, are all around the display of the ```<textarea id="display" value={equation} >```  The failures show an output of '' But in each case there is an output on the screen rendered to the that textarea. There is an initial 0 in the equations but tests fail because of an empty string value. Specifically
1. Test 7 Element with with id="display" should show 0 : expected '' to equal '0'. 
    * ***Only 0 displays on the screen with every push of the associated button marked C.***
2. Test 8: Numbers do not display correctly within id="display" : expected '' to equal '123' 
    * ***They seem to display correctly and as expected. Unless the initial 0 is a problem.***
3. Test 9 The expression 3 + 5 * 6 - 2 / 4 should produce 32.5 or 11.5 as an. 
    *  ***It renders the answer 32.5***
4. Test 10 An input of 0 0 0 should display 0 : expected '' to equal '0'
    * ***Entering 0 0 0 renders a display of 0***
5. Test 11: An input of "5 . . 0" should display 5.0 : expected '' to equal '5.0'
    * ***Entering "5..0" displays "05.0" The failure states expected '' to equal '5.0'. The display is not ''***
6. Test 12: The expression "10.5 - 5.5" should produce an output of "5" : expected '' to equal '5'
    * ***This one should be a failure. As of now, there is an error that I need to fix.   Fixed now***
7. Test 13: The sequence "5 * - 5" = should produce an output of "-25" : expected '' to equal '-25'
    * ***Entering "5 * - 5* renders an  output of -25. Showed this in the screen shot below which shows the dev tools elements showing the correct id and value.***
8. Test 14: The sequence "5 - 2 = / 2 =" should produce an output of "1.5" : expected '' to equal '1.5'
    * ***Entering " 5 - 2 = / =" produces the output 1.5***
9. Test 15:  The expression "2 / 7" should produce an output number with at least 4 decimal places of precision : expected false to be truthy
    * ***Entering " 2 / 7" produces the output 0.2857142857142857***

#### Browser Information

I am working on Codepen 

- Browser Name, Version: Version 80.0.3987.132 (Official Build) (64-bit)
- Operating System: Windows
- Mobile, Desktop, or Tablet: Desktop

#### Your Code / Link to Your Pen

https://codepen.io/mpete/pen/OJMxEjy?editors=1111

```JavaScript
function handleChange(event) {
        event.target.value = equation
    }
```

```JavaScript
//...
return (
        <div className=" container">
            <div className="calculator">
                <textarea className='screen' id="display" value={equation} onChange={handleChange}></textarea>
                <div className="keypad">
//...
```

#### Screenshot

![image](https://user-images.githubusercontent.com/37202290/86086459-0ebabd00-ba5f-11ea-81a0-74067cc574ea.png)
![image](https://user-images.githubusercontent.com/37202290/86300362-da124700-bbbf-11ea-8e6d-d3b860385772.png)

**Comment by MikePia** [comment]
Fixed my bug for Test 12. I believe it should also be a pass now.

**Comment by moT01** [comment]
For more visibility, please open issues on the [main repo](https://github.com/freeCodeCamp/freeCodeCamp/issues) for these. Thanks and happy coding 🎉

**Comment by MikePia** [comment]
Thank you for looking into this but what happened? Did my code  not pass the tests for good reason? Did you find something in the testing code? There still seem to be errors in the tests.  Mike Petersen From: TomSent: Friday, September 18, 2020 6:02 PMTo: freeCodeCamp/testable-projects-fccCc: MikePia; AuthorSubject: Re: [freeCodeCamp/testable-projects-fcc] I believe my calculator should pass all (but one just discovered bug) tests but only 7 of 16 pass. (#1179) Closed #1179.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by nhcarrigan** [comment]
Hey there @MikePia - Taking a look into the code that the tests rely on, I see that the tests call a `getInputValue` function on the `#display` element. 

That `getInputValue` function checks if the element is an `<input>` element. If it is, the test looks for the `.value`, otherwise it looks for the `.innerText`. Some of the errors you face come from using a `<textarea>` element, which like the `<input>` element, has a `.value` instead of an `.innerText` property.

The other errors, I believe, might be coming from two things:
* The initial zero is not cleared when I click a button
* There are spaces between the numbers and operators
